internal/venice-common/src/main/java/com/linkedin/venice/helix/StoragePersonaRepository.java

```diff
@@ -78,15 +78,18 @@ public boolean hasPersona(String name) {
   /**
    * Clears the cache and re-loads the cache data from Zookeeper.
    */
-  public void refresh() {
+  public final void refresh() {
     personaLock.lock();
-    List<StoragePersona> personas = storagePersonaAccessor.getAllPersonasFromZk();
-    personaMap = new VeniceConcurrentHashMap<>();
-    storeNamePersonaMap = new VeniceConcurrentHashMap<>();
-    for (StoragePersona persona: personas) {
-      cachePersona(persona);
+    try {
+      List<StoragePersona> personas = storagePersonaAccessor.getAllPersonasFromZk();
+      personaMap = new VeniceConcurrentHashMap<>();
+      storeNamePersonaMap = new VeniceConcurrentHashMap<>();
+      for (StoragePersona persona: personas) {
+        cachePersona(persona);
+      }
+    } finally {
+      personaLock.unlock();
```

Comment:
Good suggestion but `AutoCloseableLock` uses `lock` instead of `ReentrantLock`. I think we can keep it this way for now.

Reply:
It is fine to keep it this way for now. In case that you want to find an example for this case, the following might be a good one.

```
  private void addPersona(StoragePersona persona) {
    try (AutoCloseableLock ignore = AutoCloseableSingleLock.of(personaLock)) {
      storagePersonaAccessor.createPersona(persona);
      cachePersona(persona);
    }
  }
```